frontend/trpc/routes/documents/internal.ts

```
@@ -0,0 +1,124 @@
+
+export const internalDocumentsRouter = createRouter({
+  // Upload PDF and create document
+  upload: companyProcedure
+    .input(z.object({
+      name: z.string(),
+      file: z.string(), // base64 encoded PDF
+      isTemplate: z.boolean().default(false),
+      type: z.number(), // DocumentType enum
+    }))
+    .mutation(async ({ ctx, input }) => {
+      const fileStorageKey = `internal-${Date.now()}-${input.name.replace(/[^a-zA-Z0-9.-]/g, '_')}.pdf`;
```

ESLint

[error] Use the 'u' flag.

(require-unicode-regexp)